src/libraries/System.Net.Security/src/System/Net/Security/SslStream.IO.cs

```diff
@@ -120,9 +125,29 @@ private Task ProcessAuthenticationAsync(bool isAsync = false, CancellationToken
 
         private async Task ProcessAuthenticationWithTelemetryAsync(bool isAsync, CancellationToken cancellationToken)
         {
-            NetSecurityTelemetry.Log.HandshakeStart(IsServer, _sslAuthenticationOptions.TargetHost);
-            long startingTimestamp = Stopwatch.GetTimestamp();
+            long startingTimestamp;
+            if (NetSecurityTelemetry.Log.IsEnabled())
+            {
+                NetSecurityTelemetry.Log.HandshakeStart(IsServer, _sslAuthenticationOptions.TargetHost);
+                startingTimestamp = Stopwatch.GetTimestamp();
+            }
+            else
+            {
+                startingTimestamp = 0;
+            }
 
+            using Activity? activity = s_activitySource.StartActivity(ActivityName);
+            if (activity is not null)
+            {
+                activity.DisplayName = IsServer ? "TLS server" : $"TLS client {TargetHostName}";
```

Comment:
Should this include "handshake"?
E.g. From your video, other activities have the action in the display name (connect, wait for connection, ...)

Reply:
[Agree](https://github.com/open-telemetry/semantic-conventions/pull/1192#discussion_r1673902817). Will change it unless there is pushback.

Reply:
no pushback :)